### PR TITLE
UI tests have been added for all the items from the page. Functionali…

### PR DESCRIPTION
**Pull request by LiviuAC**
…ty tests have been added for burger menu, back to products button, add to cart button and footer icons.

### CODE REVIEW

File: test/pageobjects/PageObject.js [hunk 1]
```diff
@@ -32,6 +32,11 @@ class PageObject {
         return browser.$$("a[class='bm-item menu-item']");
     }
 
+    get btnCloseBurgerMenu() {
+        return browser.$("button[id='react-burger-cross-btn']");
+    }
+
+
```
Comment (axboory) [inline test/pageobjects/PageObject.js]: delete extra empty row

File: test/pageobjects/ProductDetailsPage.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+    get btnBackToProducts() {
+        return browser.$("//button[@id='back-to-products']"); //
+    }
 
     get detailsImage() {
         return browser.$("//img[@class='inventory_details_img']");
     }
+
+    get detailsName() {
+        return browser.$("//div[@class='inventory_details_name large_size']");
```
Comment (axboory) [inline test/pageobjects/ProductDetailsPage.js]: try to write this xpath only with `inventory_details_name` class. without `large_size`

File: test/pageobjects/ProductDetailsPage.js [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
+    }
+
+    get detailsDescription() {
+        return browser.$("//div[@class='inventory_details_desc large_size']");
```
Comment (axboory) [inline test/pageobjects/ProductDetailsPage.js]: same as above

File: test/specs/productDetails.js [hunk 4]
```diff
@@ -1,29 +1,199 @@
...
+        it("the 'Logout' button should redirect to the Login page", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnLogout.waitForClickable();
+            await ProductsPage.btnLogout.click();
+
+            expect(await browser.getUrl()).toContain(LoginPage.url);
+        })
+    })
```
Comment (axboory) [inline test/specs/productDetails.js]: your already wrote the tests for burger menu. better create a separate file for burger menu functionality tests, and in productDetails.js and products.js check only that the burger menu button is displayed.

File: test/specs/productDetails.js [hunk 5]
```diff
@@ -1,29 +1,199 @@
...
+        })
+
+        it("the 'LinkedIn' icon should redirect to the Sauce Labs LinkedIn url", async () => {
+            await ProductsPage.linkedInIcon.waitForClickable()
+            await ProductsPage.linkedInIcon.click()
+            await browser.switchWindow("https://www.linkedin.com/company/sauce-labs/")
+
+            expect(await browser.getUrl()).toEqual("https://www.linkedin.com/company/sauce-labs/");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
```
Comment (axboory) [inline test/specs/productDetails.js]: same as the burger menu tests. you don't have to check the footer functionalities for every page, unless there are known differences between pages.

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/pageobjects/PageObject.js (hunk 1 above)
Comment (LiviuAC) [reply test/pageobjects/PageObject.js]: extra row deleted

File: test/pageobjects/ProductDetailsPage.js (hunk 2 above)
Comment (LiviuAC) [reply test/pageobjects/ProductDetailsPage.js]: the xpath has been changed into "//*[contains(@class,'inventory_details_name')]"

File: test/pageobjects/ProductDetailsPage.js (hunk 3 above)
Comment (LiviuAC) [reply test/pageobjects/ProductDetailsPage.js]: the xpath has been changed into "//*[contains(@class,'inventory_details_desc ')]"

File: test/specs/productDetails.js (hunk 4 above)
Comment (LiviuAC) [reply test/specs/productDetails.js]: the file burgerMenu has been added into the specs folder and the modifications have been made in products and productDetails

File: test/specs/productDetails.js (hunk 5 above)
Comment (LiviuAC) [reply test/specs/productDetails.js]: the file footerIconsFunctionalityTests has been added into the specs folder and the modifications have been made in products and productDetails

File: test/specs/burgerMenu.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    it("the 'Logout' button should redirect to the Login page", async () => {
+        await LoginPage.open();
+        await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+        await ProductsPage.btnBurgerMenu.click();
+        await ProductsPage.btnLogout.waitForClickable();
+        await ProductsPage.btnLogout.click();
+
+        expect(await browser.getUrl()).toContain(LoginPage.url);
+    })
+})
```
Comment (axboory) [inline test/specs/burgerMenu.js:67]: add new line at the end of file

File: test/specs/footerIconsFunctionalityTests.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    it("the 'LinkedIn' icon should redirect to the Sauce Labs LinkedIn url", async () => {
+        await ProductsPage.linkedInIcon.waitForClickable()
+        await ProductsPage.linkedInIcon.click()
+        await browser.switchWindow("https://www.linkedin.com/company/sauce-labs/")
+
+        expect(await browser.getUrl()).toEqual("https://www.linkedin.com/company/sauce-labs/");
+        await ProductsPage.closeWindowAndSwitchBack()
+    })
+})
```
Comment (axboory) [inline test/specs/footerIconsFunctionalityTests.js:42]: add newline at the end of file

**Pull request merged**